Source/WebCore/PAL/pal/spi/cocoa/WritingToolsSPI.h

```diff
@@ -34,25 +37,170 @@ DECLARE_SYSTEM_HEADER
 #import <WritingTools/WTSession_Private.h>
 #import <WritingTools/WritingTools.h>
 
-#if PLATFORM(MAC)
+#else
 
-using PlatformWritingToolsBehavior = NSWritingToolsBehavior;
+#import <Foundation/Foundation.h>
 
-constexpr auto PlatformWritingToolsBehaviorNone = NSWritingToolsBehaviorNone;
-constexpr auto PlatformWritingToolsBehaviorDefault = NSWritingToolsBehaviorDefault;
-constexpr auto PlatformWritingToolsBehaviorLimited = NSWritingToolsBehaviorLimited;
-constexpr auto PlatformWritingToolsBehaviorComplete = NSWritingToolsBehaviorComplete;
+extern NSAttributedStringKey const WTWritingToolsPreservedAttributeName;
 
-using PlatformWritingToolsResultOptions = NSWritingToolsResultOptions;
+typedef NS_ENUM(NSInteger, WTRequestedTool) {
+    WTRequestedToolIndex = 0,
 
-constexpr auto PlatformWritingToolsResultPlainText = NSWritingToolsResultPlainText;
-constexpr auto PlatformWritingToolsResultRichText = NSWritingToolsResultRichText;
-constexpr auto PlatformWritingToolsResultList = NSWritingToolsResultList;
-constexpr auto PlatformWritingToolsResultTable = NSWritingToolsResultTable;
+    WTRequestedToolProofread = 1,
+    WTRequestedToolRewrite = 2,
+    WTRequestedToolRewriteProofread = 3,
+
+    WTRequestedToolRewriteFriendly = 11,
+    WTRequestedToolRewriteProfessional = 12,
+    WTRequestedToolRewriteConcise = 13,
+    WTRequestedToolRewriteOpenEnded = 19,
+
+    WTRequestedToolTransformSummary = 21,
+    WTRequestedToolTransformKeyPoints = 22,
+    WTRequestedToolTransformList = 23,
+    WTRequestedToolTransformTable = 24,
+
+    WTRequestedToolSmartReply = 101,
+
+    WTRequestedToolCompose = 201,
```

Comment:
I think we're not referring to most of these symbols directly, right? Do you want to pare down the list? And a couple are only used in API tests, so maybe those can just be bespoke forward declared in `Tools/TestWebKitAPI/Tests/WebKitCocoa/WritingTools.mm`?

Reply:
in a subsequent PR, I will actually be using all these (since rn I'm doing things in a not type safe way); I just haven't gotten around to it yet